src/platform/forms-system/src/js/components/FormNav.jsx

```diff
@@ -98,6 +100,43 @@ export default function FormNav(props) {
     : `Step ${currentChapterDisplay} of ${chaptersLengthDisplay}: ${chapterName ||
         ''}`;
 
+  const handleFocus = async () => {
+    let root = document.querySelector('#react-root');
+    if (formConfig.v3SegmentedProgressBar) {
+      // Need to provide shadowRoot for focusing on shadow-DOM elements
+      const shadowHost = await querySelectorWithShadowRoot(
+        'va-segmented-progress-bar',
+      );
+      root = shadowHost.shadowRoot;
+    }
+
+    return () => {
+      // Check main toggle to enable custom focus; the unmounting of the page
+      // before the review & submit page may cause the customScrollAndFocus
+      // function to be called inadvertently
+      if (
+        !(
+          page.chapterKey === 'review' ||
+          window.location.pathname.endsWith('review-and-submit')
+        )
+      ) {
+        if (formConfig.useCustomScrollAndFocus) {
+          customScrollAndFocus(page.scrollAndFocusTarget, index);
+        } else {
+          waitForRenderThenFocus(defaultFocusSelector, root, 400);
```

Comment:
maybe we still need to fallback to 'h2'. especially some custom pages. I think we need to test several kinds of custom pages to be sure.
couldn't we still use focusByOrder here? now that we're awaiting the progress bar, just wondering if we can avoid this extra 400ms time out

Reply:
Yes, just changed back to focusByOrder.